### PR TITLE
Switch to more robust way to find the tab from the DB in Competition Tabs controller

### PR DESCRIPTION
**Pull request by simonkellly**
Fixes issues mentioned in slack

### CODE REVIEW

File: WcaOnRails/app/controllers/competition_tabs_controller.rb [hunk 1]
```diff
@@ -43,12 +43,13 @@ def update
 
   def destroy
     @competition = competition_from_params
-    CompetitionTab.find_by(id: params[:id])&.destroy
+    CompetitionTab.find_by(id: params[:id], competition: @competition)&.destroy
```
Comment (dunkOnIT) [inline WcaOnRails/app/controllers/competition_tabs_controller.rb:46]: Is there a reason we aren't using `find_by!` here?

**Review by dunkOnIT — CHANGES_REQUESTED** [review]
My understanding of the change is we specify comp id as well as tab id so that it throws an error if the tab id doesn't belong to the comp id. 

If my understanding is correct, LGTM - just think we should change to `find_by!` on line 46.

File: WcaOnRails/app/controllers/competition_tabs_controller.rb (hunk 1 above)
Comment (simonkellly) [reply WcaOnRails/app/controllers/competition_tabs_controller.rb:46]: From what I looked at, having find_by! would throw an exception, and here we are already accounting for it being potentially null with the & so I thought not having it throw in this case was closer to the original implementation

Comment (dunkOnIT) [reply WcaOnRails/app/controllers/competition_tabs_controller.rb:46]: Ah I see! In this case I'm partial to raising an exception as that's more consistent with our other implementations, and this is a case where an exception seems appropriate, but I can see the argument both ways.

Comment (simonkellly) [reply WcaOnRails/app/controllers/competition_tabs_controller.rb]: Reasonable, changes made in latest commit

**Comment by simonkellly** [comment]
I think the tests failures are unrelated to the changes

**Review by dunkOnIT — APPROVED** [review]
LGTM! Agree the test failures look unrelated, have re-run the job and will look to merge once they pass

File: WcaOnRails/app/controllers/competition_tabs_controller.rb [hunk 2]
```diff
@@ -27,12 +27,12 @@ def create
 
   def edit
     @competition = competition_from_params
-    @competition_tab = CompetitionTab.find(params[:id])
+    @competition_tab = CompetitionTab.find_by!(id: params[:id], competition: @competition)
```
Comment (gregorbg) [inline WcaOnRails/app/controllers/competition_tabs_controller.rb:30]:
```suggestion
    @competition_tab = @competition.tabs.find(params[:id])
```

Comment (gregorbg) [reply WcaOnRails/app/controllers/competition_tabs_controller.rb:30]: `find` expects an ID as its only parameter. If you access the tabs through the `has_many :tabs` association in the `Competition` model, you can let Rails handle the DB loading via caches and stuff and don't have to care about manually retrieving the tab yourself.

File: WcaOnRails/app/controllers/competition_tabs_controller.rb [hunk 3]
```diff
@@ -43,12 +43,13 @@ def update
 
   def destroy
     @competition = competition_from_params
-    CompetitionTab.find_by(id: params[:id])&.destroy
+    CompetitionTab.find_by!(id: params[:id], competition: @competition).destroy
```
Comment (gregorbg) [inline WcaOnRails/app/controllers/competition_tabs_controller.rb:46]:
```suggestion
    @competition.tabs.find(params[:id]).destroy
```

File: WcaOnRails/app/controllers/competition_tabs_controller.rb [hunk 4]
```diff
@@ -43,12 +43,13 @@ def update
...
     redirect_to competition_tabs_path(@competition)
   end
 
   def reorder
-    competition_tab = CompetitionTab.find(params[:id])
+    competition = competition_from_params
+    competition_tab = CompetitionTab.find_by!(id: params[:id], competition: competition)
```
Comment (gregorbg) [inline WcaOnRails/app/controllers/competition_tabs_controller.rb:52]:
```suggestion
    competition_tab = competition.tabs.find(params[:id])
```

File: WcaOnRails/app/controllers/competition_tabs_controller.rb (hunk 2 above)
Comment (simonkellly) [reply WcaOnRails/app/controllers/competition_tabs_controller.rb]: Very nice way of handling it 😎

**Review by gregorbg — APPROVED** [review]

**Pull request merged**